app/controllers/v0/messages_controller.rb

```diff
@@ -1,9 +1,19 @@
 # frozen_string_literal: true
 module V0
   class MessagesController < SMController
+    include Filterable
+
+    PERMITTED_FILTERS = {
+      'subject' => %w(eq not_eq),
+      'sender_name' => %w(eq not_eq),
+      'sent_date' => %w(eq lteq gteq)
+    }.freeze
+
     def index
       resource = client.get_folder_messages(params[:folder_id].to_s)
       raise VA::API::Common::Exceptions::RecordNotFound, params[:folder_id] unless resource.present?
+
+      resource = can_filter?(Message, PERMITTED_FILTERS) ? resource.find_by(params[:filter]) : resource
```

Comment:
this is good for now, but we're going to clean this up in a separate PR so that the controller doesn't have to be concerned with it anymore. but 👍